modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/RebalanceRaftGroupEventsListener.java

```diff
@@ -121,14 +122,31 @@ public RebalanceRaftGroupEventsListener(
 
     /** {@inheritDoc} */
     @Override
-    public void onLeaderElected() {
-        // TODO: IGNITE-16800 implement this method
+    public void onLeaderElected(long term) {
```

Comment:
As was discussed, I believe that we should evaluate onLeaderElected() and onNewPeersConfigurationApplied() in separate rebalance executor instead raft thread.